### PR TITLE
unit-test for replace keys 

### PR DESCRIPTION
**Pull request by allenhan2**
## Change Description

Add unit test for PR: https://github.com/EOSIO/eos/pull/8929

## Consensus Changes
- [ ] Consensus Changes

## API Changes
- [ ] API Changes

## Documentation Additions
- [ ] Documentation Additions

### CODE REVIEW

File: unittests/chain_tests.cpp [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+using namespace eosio::chain;
+using namespace eosio::testing;
+
+BOOST_AUTO_TEST_SUITE(chain_tests)
+
+BOOST_AUTO_TEST_CASE( replace_producer_keys ) try {
+   validating_tester tester;
+
+   const auto head_ptr = tester.control->head_block_state();
+   BOOST_ASSERT(head_ptr);
```
Comment (heifner) [inline unittests/chain_tests.cpp:20]: All these `BOOST_ASSERT` should be `BOOST_REQUIRE`

File: unittests/chain_tests.cpp [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+         BOOST_REQUIRE_EQUAL(key.key, new_key);
+         BOOST_REQUIRE_EQUAL(key.weight, expected_key_weight);
+       }
+   }
+} FC_LOG_AND_RETHROW()
+
+BOOST_AUTO_TEST_CASE( replace_account_keys ) try {
+   validating_tester tester;
+   const name usr("eosio");
+   const name active_permission("active");
```
Comment (heifner) [inline unittests/chain_tests.cpp:56]: Use `eosio::chain::config::system_account_name` & `eosio::chain::config::active_name`

File: unittests/chain_tests.cpp [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+   const auto& rlm = tester.control->get_resource_limits_manager();
+   const auto* perm = tester.control->db().find<permission_object, by_owner>(boost::make_tuple(usr, active_permission));
+   BOOST_ASSERT(perm != NULL);
+
+   const int64_t old_size = (int64_t)(chain::config::billable_size_v<permission_object> + perm->auth.get_billable_size());
```
Comment (heifner) [inline unittests/chain_tests.cpp:61]: Why the cast to `int64_t` here and below?

File: unittests/chain_tests.cpp (hunk 1 above)
Comment (allenhan2) [reply unittests/chain_tests.cpp:20]: I can change to BOOST_REQUIRE.

File: unittests/chain_tests.cpp (hunk 3 above)
Comment (allenhan2) [reply unittests/chain_tests.cpp:61]: it is used in replace_account_keys  implementation, https://github.com/EOSIO/eos/pull/8929/files

File: unittests/chain_tests.cpp (hunk 2 above)
Comment (allenhan2) [reply unittests/chain_tests.cpp]: updated

File: unittests/chain_tests.cpp (hunk 3 above)
Comment (heifner) [reply unittests/chain_tests.cpp:61]: Yuck, but ok.

**Pull request merged**